### PR TITLE
feat: convert mdx components in web app to mdx-components package

### LINKED ISSUE #1103: Convert MDX components in web-app to `mdx-components` package
🦖@alisonjoseph
🐳 @francinelucca 
🐯@jdharvey-ibm
🐢 @andreancardona 

## Markdown components

- [x] h1: H1 🐳 
- [x] h2: H2 🐳 
- [x] h3: H3 🐳 
- [x] h4: H4 🐳 
- [x] h5: H5 🐳 
- [x] h6: H6 🐳 
- [x] p: P 🐳 
- [x] ol: OL 🐢 
- [x] li: LI 🐢 
- [x] #1457
- [x] a: Link 🐳 
- [x] blockquote: Blockquote 🐳
- [x] table: PageTable 🐳 
- [x] pre: Code 🐳 -- ⚠️ SB `[Object object]` issue

##  Core Carbon components

Q: Why are these here? Why are the carbon components as-is not sufficient to host our carbon documentation and other MDX? Can these be removed in favor of the base carbon components?

- [x] AccordionItem
- [x] Button
- [x] StructuredListWrapper
- [x] StructuredListHead
- [x] StructuredListRow
- [x] StructuredListCell
- [x] StructuredListBody
- [x] UnorderedList
- [x] ListItem
- [x] Tag

## Gatsby theme components

- [x] Accordion 🦖
- [x] AnchorLink 🐯
- [x] AnchorLinks 🐯
- [x] ArtDirection 🦖
- [x] ArticleCard  🐳🐢
- [x] Aside 🦖
- [x] AutolinkHeader 🐳 
- [x] Caption 🦖
- [x] CardGroup 🦖
- [x] ColorBlock🐯
- [x] Column 🦖
- [x] DoDont 🦖
- [x] DoDontRow 🦖
- [x] Divider 🦖
- [x] GifPlayer 🐳 
- [x] Grid 🦖
- [x] ImageWrapper 🐳 
- [x] InlineNotification 🐢
- [x] MiniCard 🐳🐢
- [x] PageDescription 🐳 
- [x] Preview 🐢 
- [x] Profile 🐳 
- [x] ResourceCard 🐳  
- [x] Row 🦖
- [x] StorybookDemo 🦖 -- ⚠️ lint error needs fixing -- Fixed!
- [x] Tab 🐯
- [x] Tabs 🐯
- [x] Title🐯
- [x] Video🐯

### Internal Components
- [x] ColorGrid 🐢 
- [x] ColorPalette 🐢 
- [ ] StatusIndicatorTable -- ⚠️ need help with '/status-icon' svgs import
- [x] TypeScaleTable 🦖
- [x] TypeWeight 🦖

### Deferred
-  img: Image -- ⚠️ Deferred
- Image -- ⚠️ Deferred
- MdxIcon 🐳  -- ⚠️ Deferred

### Convert to js pages (https://github.com/carbon-design-system/carbon-platform/issues/1446)
- ColorTokenTable 
- Glossary
- IconLibrary
-  PictogramLibrary -
- TypesetStyle

### LINKED ISSUE #1384: Web-App: CodeSnippet styling issues
**Description**

- [x] CodeSnippet usage along with text in a paragraph is overlapping eachother

image

- [x] CodeSnippet is overflowing content in small sized screens 
image

**Steps to reproduce**

1. Drop these contents into a `test.mdx` file in `services/web-app/pages`
```mdx
Inline styles Inline styles in the form of string: `style="padding-top: 20px; color: red"` instead
  of javascript objects: `style={{paddingTop: '20px', color: 'red'}}` will result in a Component not
  rendering error.
```
2. Navigate to http://localhost/test
3. Make screen size width 320

Note: you can also reproduce this in a `*.js` file:
```jsx
  <div>
              Inline styles in the form of string:{' '}
              <CodeSnippet type="inline">{'style="padding-top: 20px; color: red"'}</CodeSnippet>{' '}
              instead of javascript objects:
              <CodeSnippet type="inline">
                {"style={{ paddingTop: '20px', color: 'red' }}"}
              </CodeSnippet>{' '}
              will result in a Component not rendering error.
    </div>
```

### LINKED ISSUE #1413: Remove YAML loader in favor of JS modules
This issue is for removing the `withYaml` loader from the next config file.

This looks to only be used in one place right now, which is to use `services/web-app/data/status-indicators/status-indicators.yaml` for the status indicators table component. This should be removed in favor of a JS file defining the status indicators, which can be checked/linted.

### LINKED ISSUE #1496: Finalize list of one-use components from mdx-components package
Continue with the mural investigation to figure out which components are only used in one place (and hopefully not in any dynamic docs) and add those to the list of components that will be excluded from mdx-components package.

https://app.mural.co/t/ibm14/m/ibm14/1667230506318/3c007d2b56bfc0b1c820e15d7d946285da5ae4a2?sender=jdharvey8136

Once this list is done, the conversion PR can be updated accordingly to "revert" changes we no longer want.

#1423

### PR DESCRIPTION
**Pull request by jharvey10**
Closes #1103
Closes #1384 
Closes #1413 
Closes #1496 

### Things to check

- [x] Did we remember to remove the components from components.js?
- [x] Did we remember to add each component to `mdx-components/src/main/index.ts`?
- [x] Did we remember to add the styles for each component to `mdx-components/src/main/index.scss`?
- [x] Did we remember to move over the stories files?
- [x] Did we remove all components that are re-exports of carbon ones in favor if the core carbon version?
- [x] Did we remember to update the unit test snapshots?
- [x] Did we remember to export the typescript props interface from each component (and export it from index.ts)?

### Open questions

- [x] What do we do about the `pages/mdx` files. Should they be roped into the storybook stories in the mdx-components package? (#1499)

### Testing / reviewing

Compare the component visuals and functionality to what they were before the conversion.

### CODE REVIEW

**Comment by mattrosno** [comment]
@jdharvey-ibm @alisonjoseph we shouldn't need this page, once all of these MDX components are documented in individual Storybook stories.

https://next.carbondesignsystem.com/mdx

**Comment by alisonjoseph** [comment]
@mattrosno @jdharvey-ibm Let's wait until everything looks good and is approved before removing the test mdx pages, so we can review things in the site easily and compare to live.

File: packages/mdx-components/src/main/card-group/card-group.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * LICENSE file in the root directory of this source tree.
+ */
+import { Column, Grid } from '@carbon/react'
+import { clsx } from 'clsx'
+import PropTypes from 'prop-types'
+import React, { ReactNode } from 'react'
+
+// TODO: ask Joe - how can we use this in here?
+// import { mediaQueries, useMatchMedia }
+// from '../../../../services/web-app/utils/use-match-media.js'
```
Comment (alisonjoseph) [inline packages/mdx-components/src/main/card-group/card-group.tsx:14]: @jdharvey-ibm tagging you now so I don't forget to ask you next week

Comment (jdharvey-ibm) [reply packages/mdx-components/src/main/card-group/card-group.tsx:14]: @alisonjoseph There's no easy way for us to use this in both places :/

I think core Carbon does have a media query hook that we could get at. That could potentially be used here instead.

If not, our best option is to copy-paste those two imports from the web-app into the mdx-components package.

File: packages/mdx-components/src/main/storybook-demo/storybook-demo.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+import Caption from '../caption/caption.js'
+import { MdxComponent } from '../interfaces.js'
+import { withPrefix } from '../utils.js'
+
+/* eslint-disable  @typescript-eslint/no-explicit-any  -- TODO ask Joe how to fix this */
+type Variants = { [key: string]: any }
+/* eslint-enable @typescript-eslint/no-explicit-any  -- Re-enable rule */
+
+interface StorybookDemoProps {
```
Comment (alisonjoseph) [inline packages/mdx-components/src/main/storybook-demo/storybook-demo.tsx:21]: @jdharvey-ibm can you take a look at this file when you get a chance 😅

File: packages/mdx-components/src/main/storybook-demo/storybook-demo.tsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    [withPrefix('wide')]: wide
+  })
+
+  const [theme, setTheme] = useState(themeItems[0]!.src)
+  const onThemeChange = (item: { selectedItem: { src: React.SetStateAction<string> } }) => {
+    setTheme(item.selectedItem.src)
+  }
+
+  const multipleVariants = variants?.length > 1
+  console.log(' 🐬 ~ variants', variants)
```
Comment (alisonjoseph) [inline packages/mdx-components/src/main/storybook-demo/storybook-demo.tsx:80]:
```suggestion
```

**Review by francinelucca — APPROVED** [review]

**Comment by francinelucca** [comment]
we removed most prop warnings @jdharvey-ibm and the ones left behind are reasonable!

**Pull request merged**